### DISCUSSION TITLE
How to develop SAR Matrices

### DISCUSSION THREAD

**Discussion by syedzayyan** [rdkit/rdkit#6793] [General] [answered]
Anyone here developed a SAR Matrix and conducted a Free Wilson Analysis before as described in this paper?

https://pubs.acs.org/doi/pdf/10.1021/acsomega.9b00595

**Comment by bp-kelley** [comment, marked as answer]
In the Contrib directory there is a free Wilson implementation.  It’s also described well in this blog, it requires scikit learn to operate.


https://iwatobipen.wordpress.com/2022/10/21/fw-analysis-make-easily-with-rdkit-contrib-fw-package-rdkit-chemoinformatics-memo/

if you need any help, let me know, I wrote the contribution.

**Reply by syedzayyan** [reply]
Thanks for the blog!

**Reply by syedzayyan** [reply]
An additional question. Do you think ML-based methods like RF are better than Free Wilson or something like this https://pubs.acs.org/doi/10.1021/ci5000483 ?

**Reply by bp-kelley** [reply]
FreeWilson is generally a way to "fill in the holes" from a SAR analysis and allow you to see what compounds you should make that are rearrangements from existing molecules.

I far prefer gaussian processes to random forest, you can see some examples from last years UGM:

https://github.com/rdkit/UGM_2022/blob/main/Notebooks/Kelley_Gaussian_Process_Chemistry_Kernels.ipynb

**Reply by syedzayyan** [reply]
Thanks for the reply!

Essential for my target, Adenosine A1, I am analysing the ChEMBL-filtered database. Having a large database in hand I wanted to see what parts of a molecule are important. I have tried matched molecule pairs and scaffold networks and such. Now getting into SAR matrices which.....and there is no definitive thing part that seems particularly important. I am taking solace in the fact that these compounds are for a GPCR and GPCRs are weird. :(
